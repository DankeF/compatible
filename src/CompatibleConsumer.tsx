import { ConfigProvider } from 'antd';
import { ConfigConsumer, ConfigConsumerProps } from 'antd/lib/config-provider';

export { ConfigConsumerProps };

const MergedConfigConsumer =
  ConfigProvider?.ConfigContext?.Consumer || ConfigConsumer;

export default MergedConfigConsumer;
