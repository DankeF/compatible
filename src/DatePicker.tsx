import * as React from 'react';
import { DatePicker } from 'antd';

import warning from 'rc-util/lib/warning';

type FCProps<Component> = Component extends React.ComponentType<infer Props> ? Props : never;
type InferProps = FCProps<typeof DatePicker>;

type CompatibleDatePickerProps = InferProps & {
  /** @deprecated Please use `popupClassName` instead. */
  dropdownClassName?: string;
  popupClassName?: string;
}

const CompatibleDatePicker = React.forwardRef(
  (
    { dropdownClassName, popupClassName,...restProps }: CompatibleDatePickerProps,
      ref
  ) => {
    warning(
      !dropdownClassName,
      "[antd: DatePicker] `dropdownClassName` is removed in v5, please use `popupClassName` instead.",
    );

    return (
      <DatePicker {...restProps} popupClassName={popupClassName || dropdownClassName} ref={ref} />
    );
  },
);


CompatibleDatePicker.RangePicker = React.forwardRef(
  (
    { dropdownClassName, popupClassName,...restProps }: CompatibleDatePickerProps,
    ref
  ) => {
    console.log('dropdownClassName--->>>',dropdownClassName)
    warning(
      !dropdownClassName,
      "[antd: RangePicker] `dropdownClassName` is removed in v5, please use `popupClassName` instead.",
    );

    return (
      <DatePicker.RangePicker {...restProps} popupClassName={popupClassName || dropdownClassName} ref={ref} />
    );
  },
);

if (process.env.NODE_ENV !== 'production') {
  CompatibleDatePicker.displayName = 'CompatibleDatePicker';
}

export default CompatibleDatePicker;
