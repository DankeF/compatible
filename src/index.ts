export { default as Icon } from './icon';
export { default as Form } from './form';
