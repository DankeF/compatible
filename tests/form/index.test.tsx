import React from 'react';
import { render } from 'enzyme';
import { Input } from 'antd';
import { Form } from '../../src';
import { FormComponentProps } from '../../src/form';

describe('Form', () => {
  it('should render correctly', () => {
    class MyForm extends React.Component<FormComponentProps> {
      render() {
        return (
          <Form>
            <Form.Item>
              {this.props.form.getFieldDecorator('name')(<Input />)}
            </Form.Item>
          </Form>
        );
      }
    }

    const WrappedForm = Form.create()(MyForm);

    const wrapper = render(<WrappedForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
