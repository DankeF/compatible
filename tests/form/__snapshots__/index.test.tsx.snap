// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
<form
  class="ant-legacy-form ant-legacy-form-horizontal"
>
  <div
    class="ant-row ant-legacy-form-item"
  >
    <div
      class="ant-col ant-legacy-form-item-control-wrapper"
    >
      <div
        class="ant-legacy-form-item-control"
      >
        <span
          class="ant-legacy-form-item-children"
        >
          <input
            class="ant-input"
            data-__field="[object Object]"
            data-__meta="[object Object]"
            id="name"
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
</form>
`;
